tutorials/grid_coordinates.py

```diff
@@ -0,0 +1,208 @@
+"""
+.. _grid_coordinates:
+
+Gridding coordinates
+====================
+
+Grid coordinates in Verde are used to create points on a regularly spaced grid
+that are  then used in the spline method to interpolate between sample data
+points. The grid can be specified either by the number of points in each
+dimension (the *shape*) or by the grid node spacing.
+
+Creating regular grids in Verde is done using the :func:`verde.grid-coordinates`
+function. The function uses :func:`numpy.linspace` and
+:func:`numpy.meshgrid` to interpolate between points in both the west-east and
+south-north directions to create a two-dimensional spatial grid. To this
+end, the function will adjust either the region bounds or the spacing between
+grid nodes.
+
+First let's create a region that is 1555 units west-east and 1250 units south-
+north, and we will set an initial spacing to 500 units. 
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import verde as vd
+from matplotlib.patches import Rectangle
+
+# create some bounds for the region and set the spacing for the grid points
+spacing = 500
+west = 0
+east = 1555
+south = 0
+north = 1250
+region = (west, east, south, north)
+
+########################################################################################
+# How Grid Coordinates works
+# --------------------------
+# When we choose to adjust the region the function takes the original region, and
+# divides the south-north and west-east extents of the region by the spacing to
+# get the number of grid points in both directions. This value is then
+# rounded and converted to an integer. The function then adds one to the integer
+# to get the number of nodes rather than segments.The function then multiplies
+# the number of grid points by the chosen spacing to create the new region.
```

Comment:
I don't know if explaining what exactly is going on under the hood is entirely necessary on this kind of tutorials.
Maybe would be nicer to show the problem that raises from the initial choice of region and spacing.
Something like:

```
We cannot define a set of regularly spaced points by 500 units on this region:
the range of the west-east and south-north boundaries are not multiple of 500 units.
So we must choose to change either:
- the boundaries of the region in order to fit the spacing, or
- the spacing in order to fit the region boundaries.
```

Then we can introduce the two approaches: adjust region or adjust spacing.

Reply:
I like how you reframed this section. I think it's easier to understand the difference between adjusting spacing vs boundaries.